compiler/rustc_interface/src/passes.rs

```diff
@@ -663,13 +662,11 @@ fn write_out_deps(
     }
 }
 
-pub fn prepare_outputs(
-    sess: &Session,
-    krate: &ast::Crate,
-    boxed_resolver: &RefCell<BoxedResolver>,
-    crate_name: Symbol,
-) -> Result<OutputFilenames> {
+fn output_filenames(tcx: TyCtxt<'_>, (): ()) -> Result<&Arc<OutputFilenames>> {
```

Comment:
```suggestion
fn output_filenames(tcx: TyCtxt<'_>, (): ()) -> Result<&OutputFilenames> {
```
Is `Arc` still needed here?
Does it need to be cloned to some place where `tcx` and its arenas are longer available?

Reply:
Yes, llvm, ssa and cranelift all store a clone of it it until after the TyCtxt is gone.